pkg/controller/common/license/model.go

```diff
@@ -69,7 +72,12 @@ func (l EnterpriseLicense) IsValid(instant time.Time) bool {
 
 // IsTrial returns true if this is a self-generated trial license.
 func (l EnterpriseLicense) IsTrial() bool {
-	return l.License.Type == LicenseTypeEnterpriseTrial
+	return l.License.Type == LicenseTypeEnterpriseTrial || l.License.Type == LicenseTypeLegacyTrial
+}
+
+// IsECKManagedTrial returns true if this license has been generated by trial controller or is eligible to be turned into one.
```

Comment:
Nit: "or is eligible to be turned into one." is kind of confusing, but I'm not sure what would be better. Maybe "or is eligible to have its license generated by the trial controller" but that still reads weird. Maybe just "or is eligible to be"? All sound a little bit off